package NYT.NControllerAgent.NProto;

import "yt/ytlib/controller_agent/proto/job.proto";

import "yt_proto/yt/client/node_tracker_client/proto/node.proto";

import "yt_proto/yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TAllocationInfo
{
    required NYT.NProto.TGuid allocation_id = 1;
}

// Sends job statistics to controller agent.
message TReqHeartbeat
{
    required int32 node_id = 1;
    required NNodeTrackerClient.NProto.TNodeDescriptor node_descriptor = 2;
    required NYT.NProto.TGuid controller_agent_incarnation_id = 3;

    repeated TJobStatus jobs = 4;
    repeated NYT.NProto.TGuid unconfirmed_job_ids = 9;
    required int32 confirmed_job_count = 10;

    repeated TAllocationInfo allocations = 11;
}

message TJobToAbort
{
    required NYT.NProto.TGuid job_id = 1;
    optional int32 abort_reason = 2;
}

message TJobToInterrupt
{
    required NYT.NProto.TGuid job_id = 1;
    required int64 timeout = 2;
    required int32 reason = 3; // NScheduler::EInterruptReason
}

message TJobToFail
{
    required NYT.NProto.TGuid job_id = 1;
}

message TJobToStore
{
    required NYT.NProto.TGuid job_id = 1;
}

message TJobToConfirm
{
    required NYT.NProto.TGuid job_id = 1;
}

message TRspHeartbeat
{
    repeated TJobToStore jobs_to_store = 1;
    repeated TJobToRemove jobs_to_remove = 2;
    repeated TJobToConfirm jobs_to_confirm = 3;
    repeated TJobToAbort jobs_to_abort = 4;
    repeated TJobToInterrupt jobs_to_interrupt = 6;
    repeated TJobToFail jobs_to_fail = 7;
    repeated NYT.NProto.TGuid unknown_operation_ids = 5;
}

////////////////////////////////////////////////////////////////////////////////
